src/pages/Search/Search.js

```diff
@@ -1,23 +1,61 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import config from '../../config';
 import Page from '../../components/Page';
-import HeatMap from '../../components/HeatMap';
-import ResultsTable from '../../components/ResultsTable';
 import InputForm from '../../components/InputForm';
 import Headline from '../../components/Headline';
-
+import Spinner from '../../components/Spinner';
 
 export default function Search() {
+  const { slug } = useParams();
+
+  const [data, setData] = useState([]);
+  const [loading, toggleLoading] = useState(false);
+  const [errorMessage, setError] = useState('');
+
+  let count = 0;
+  let after = '';
+  const loopData = [];
+
+  const fetchData = () => {
```

Comment:
Great, you built a recursive function. That's a nice way to execute the five fetches after each other. I would try to move this function outside the Search component.

```
function fetchData() {

}

export default function Search() {
...
}
```

Then you're forced to make clear what the function depends on. You will need to move the `setData` and `toggleLoading` to the `useEffect` and add some parameters to the `fetchData` function